### PR TITLE
MAINT: Linting

### PR DESCRIPTION
**Pull request by tlimoncelli**
I fixed many, many, lint and staticcheck warnings and errors.  Sorry for the mega-PR but its the fastest way to get them all done.

CC @tresni @papakai @blackshadev @mikenz @philhug @tlimoncelli 

Note: I haven't used the checklist feature before. Let's see if this works.  I've made a checklist of the files in this PR.  We'll check them off as people confirm them.  I may check off some myself if they are just comments.)

- [ ] OWNERS
- [x] models/record.go
- [x] pkg/acme/acme.go
- [x] pkg/js/js.go
- [x] pkg/normalize/validate.go
- [x] providers/cloudflare/cloudflareProvider.go
- [ ] providers/cscglobal/api.go
- [x] providers/cscglobal/cscglobalProvider.go
- [ ] providers/easyname/api.go
- [ ] providers/easyname/provider.go
- [x] providers/hexonet/domains.go
- [x] providers/opensrs/opensrsProvider.go
- [x] providers/route53/route53Provider.go
- [x] providers/transip/transipProvider.go

### CODE REVIEW

**Comment by tlimoncelli** [comment]
CC @tresni @papakai @blackshadev @mikenz @philhug @tlimoncelli

**Comment by blackshadev** [comment]
Hi @tlimoncelli , what can I do for you? 

If you want me to check the changes in the provider I am an maintainer (Transip): You only added a comment and it is perfect.
If you need more: could you please explain what I can do for you?

**Comment by tlimoncelli** [comment]
> Hi @tlimoncelli , what can I do for you?
> 
> If you want me to check the changes in the provider I am an maintainer (Transip): You only added a comment and it is perfect. If you need more: could you please explain what I can do for you?

Yes, verifying the change is all I needed.  (You had it easy: other people had bigger changes.)

Thanks for checking in!

**Comment by tresni** [comment]
Easyname looks good, just the one change for `Id` -> `ID`

File: providers/hexonet/domains.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ func (n *HXClient) EnsureDomainExists(domain string) error {
 		}
 	} else if code == 531 {
 		return n.GetHXApiError("Not authorized to manage dnszone", domain, r)
+		// FIXME(tlim) go-staticcheck reports:
+		// identical expressions on the left and right side of the '||' operator (SA4000)
+		// Perhaps the right side should be n.IsError() or deleted?
 	} else if r.IsError() || r.IsError() {
```
Comment (papakai) [inline providers/hexonet/domains.go:27]: Morning @tlimoncelli 

I suggest to rewrite this to

```go
} else if r.IsError() || r.IsTmpError() {
```

Probably an oversight on my end some time ago.
Find tje Reference to `IsTmpError` function [here](https://github.com/hexonet/go-sdk/blob/master/responsetemplate/responsetemplate.go#L96).

Comment (tlimoncelli) [reply providers/hexonet/domains.go]: Changed as suggested. If it passed CI, I'll mark this as done.

**Comment by tlimoncelli** [comment]
Sorry to rush folks but I'd like to do a release soon, so I'm going to merge this.  The changes are rather low-risk.  The new release will go out early next week but first we're going to use it inside Stack Overflow for a bit as we usually do.

**Pull request merged**